### PR TITLE
Softmax transiogram fit and multistart in Matrix Exponential path

### PR DESCRIPTION
**Pull request by pkofod**
@juliohm please read this carefully as it may not be what you want. I changed a few things. First, proportions can be fit as you do, but the softmax version is more straight forward imho because this way they sum to 1 by construction. The multistart in the mat exp can be removed, but when running the tests I found that even if a minimum was found it was not always the global.

Some caveats:
- now that proportions are fixed by the softmax relation there is no simple way to bound each individual proportions size. I think you did this for stability of the fit, but if you need this, then softmax is not the way to go
- I made the mat exp fit a bit more expensive since it tries several starting values. I'm not sure if this is a bottle neck in practice. You can adjust as you see fit.

### CODE REVIEW

**Comment by juliohm** [comment]
Thank you @pkofod for attempting an alternative solution with a softmax rewrite. Appreciate the time you put into this! I would like to avoid core changes to the original implementation though because it has been used in past projects with great success.

There is nothing special about  the box constraints in #85, yet we still get test failures with Optim.jl v2.x, which seems to indicate an issue on the optimization side. Can we try to debug it further over there?

I will close the PR here to avoid confusion.

**Comment by juliohm** [comment]
@pkofod I am starting to appreciate your proposed solution with softmax. I've noticed that our current implementation has several flaws and that your approach is superior.

File: src/fitting/transiograms.jl [hunk 1]
```diff
@@ -123,58 +112,55 @@ function _fit(
...
+
+  # collect all box constraints for range, lengths and logits
+  l = isnothing(proportions) ? [rₗ, lₗ..., λₗ...] : [rₗ, lₗ...]
+  u = isnothing(proportions) ? [rᵤ, lᵤ..., λᵤ...] : [rᵤ, lᵤ...]
+
+  # the matrix-exponential misfit is non-convex, so a single start can land in
+  # a poor local minimum. Run a small deterministic multistart that varies the
+  # free blocks (length scale and proportion logits) and keep the best fit.
+  lstarts = isnothing(lengths′) ? [s .* lmax for s in (0.1, 0.25, 0.5, 0.75, 0.9)] : [lₒ]
+  λstarts = isnothing(proportions) ? [ntuple(c -> b * (c - m / 2), m) for b in (-1.0, 0.0, 1.0)] : [()]
```
Comment (juliohm) [inline src/fitting/transiograms.jl]: @pkofod could you please comment on why you chose these starts for the logits? What is the intuition behind this expression?

Comment (juliohm) [reply src/fitting/transiograms.jl]: I guess it is okay to fix the initial logits to be equal to zero (i.e., uniform proportions) and only vary the start for the lengths. I did that and all tests are passing. Please let me know if you think differently.

**Comment by juliohm** [comment]
@pkofod I am reverting back to `Fminbox` in order to merge the other improvements introduced in this PR.

I will open another one with just the `Fminbox` -> `LBFGSB` change afterwards and will ping you there.
